src/Symfony/Component/RateLimiter/RateLimiterBuilder.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+use Symfony\Component\Lock\LockInterface;
+use Symfony\Component\RateLimiter\Policy\FixedWindowLimiter;
+use Symfony\Component\RateLimiter\Policy\NoLimiter;
+use Symfony\Component\RateLimiter\Policy\Rate;
+use Symfony\Component\RateLimiter\Policy\SlidingWindowLimiter;
+use Symfony\Component\RateLimiter\Policy\TokenBucketLimiter;
+use Symfony\Component\RateLimiter\Storage\StorageInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+class RateLimiterBuilder
+{
+    public function __construct(
+        private StorageInterface $storage,
+        private ?LockInterface $lock = null,
+    ) {
+    }
+
+    /**
+     * @param string               $id       Unique identifier for the rate limiter
+     * @param int                  $limit    Maximum allowed hits for the passed interval
+     * @param \DateInterval|string $interval If string, must be a number followed by "second",
+     *                                       "minute", "hour", "day", "week" or "month" (or their
+     *                                       plural equivalent)
+     */
+    public function slidingWindow(string $id, int $limit, \DateInterval|string $interval): LimiterInterface
+    {
+        return new SlidingWindowLimiter($id, $limit, self::intervalNormalizer($interval), $this->storage, $this->lock);
+    }
+
+    /**
+     * @param string               $id       Unique identifier for the rate limiter
+     * @param int                  $limit    Maximum allowed hits for the passed interval
+     * @param \DateInterval|string $interval If string, must be a number followed by "second",
+     *                                       "minute", "hour", "day", "week" or "month" (or their
+     *                                       plural equivalent)
+     */
+    public function fixedWindow(string $id, int $limit, \DateInterval|string $interval): LimiterInterface
+    {
+        return new FixedWindowLimiter($id, $limit, self::intervalNormalizer($interval), $this->storage, $this->lock);
+    }
+
+    /**
+     * @param string $id       Unique identifier for the rate limiter
+     * @param int    $maxBurst The maximum allowed hits in a burst
+     * @param Rate   $rate     The rate at which tokens are added to the bucket
+     */
+    public function tokenBucket(string $id, int $maxBurst, Rate $rate): LimiterInterface
+    {
+        return new TokenBucketLimiter($id, $maxBurst, $rate, $this->storage, $this->lock);
+    }
+
+    public function compound(LimiterInterface ...$limiters): LimiterInterface
+    {
+        return new CompoundLimiter($limiters);
+    }
+
+    public function noop(): LimiterInterface
```

Comment:
```suggestion
    public function noLimit(): LimiterInterface
```

the policy name in config for `NoLimiter` is `no_limit` - it's nice to be consistent for discoverability purposes (personally, I like the `noop` name better)

Reply:
Hmm, yeah I really like `noop` too - especially since the other methods don't include "Limit" in their name.

Let's see what others think.